napari/_qt/_qapp_model/qactions/__init__.py

```diff
@@ -0,0 +1,31 @@
+from functools import lru_cache
+from typing import Optional
+
+# Submodules should be able to import from most modules, so to
+# avoid circular imports, don't import submodules at the top level here,
+# import them inside the init_qactions function.
+
+
+@lru_cache  # only call once
+def init_qactions():
+    from itertools import chain
+
+    from napari._app_model import get_app
+
+    from ...qt_main_window import Window, _QtMainWindow
+    from ._view import VIEW_ACTIONS
+
+    # Qt-specific providers/processors
+    def _provide_window() -> Optional[Window]:
+        if _qmainwin := _QtMainWindow.current():
+            return _qmainwin._window
+
+    # update the namespace with the Qt-specific types/providers/processors
+    app = get_app()
+    store = app.injection_store
+    store.namespace = {**store.namespace, 'Window': Window}
+    store.register_provider(_provide_window)
+
+    # register actions
+    for action in chain(VIEW_ACTIONS):
```

Comment:
```suggestion
    for action in VIEW_ACTIONS:
```

Reply:
as soon as I add in #4865 that chain will need to come back

Reply:
Ok. The review was contextless. 
But then why the chain is imported in function, not globally?

Reply:
yeah, sorry about that.  I can add it globally too